recorder/src/main/cpp/perf_ctx.cc

```diff
@@ -9,6 +9,15 @@ PerfCtx::MergeSemantic PerfCtx::merge_semantic(PerfCtx::TracePt pt) {
     return static_cast<MergeSemantic>((pt >> PerfCtx::MERGE_SEMANTIC_SHIFT) & MERGE_SEMANTIC_MASK);
 }
 
+metrics::Mtr* s_m_bad_pairing;
+
+PerfCtx::ThreadTracker::ThreadTracker(Registry& _reg, ProbPct& _pct, int _tid) :
+    reg(_reg), pct(_pct), ignore_count(0), effective_start(0), effective_end(0), record(false), tid(_tid) {
+    effective.reserve((MAX_NESTING * (MAX_NESTING + 1)) / 2);
+}
+
+PerfCtx::ThreadTracker::~ThreadTracker() {}
+
 void PerfCtx::ThreadTracker::enter(PerfCtx::TracePt pt) {
```

Comment:
Original source: http://osp.mans.edu.eg/cs212/FF_Edge_Triggered.htm

In context of any state change, edge-triggered means when it goes from one class of state to the other and vice-versa.

Eg. a socket not having anything for anyone to read to it having something is an edge event. A socket having 10 bytes available to be read and getting another 100, making it 110 is a level-event. 

Same definitions holds for interrupts etc (on system side).

In this case, we are transitioning from no-overflow to overflow of magnitude 1, that is the edge event, when it goes from 1 -> 2 or from 2 -> 3, or from 3 -> 2, those are all level events.

Makes sense?